src/Menu.tsx

```diff
@@ -378,20 +378,27 @@ const Menu = React.forwardRef<MenuRef, MenuProps>((props, ref) => {
     setMergedActiveKey(undefined);
   });
 
-  useImperativeHandle(ref, () => ({
-    list: containerRef.current,
-    focus: options => {
-      const shouldFocusKey =
-        mergedActiveKey ?? childList.find(node => !node.props.disabled)?.key;
-      if (shouldFocusKey) {
-        containerRef.current
-          ?.querySelector<HTMLLIElement>(
-            `li[data-menu-id='${getMenuId(uuid, shouldFocusKey as string)}']`,
-          )
-          ?.focus?.(options);
-      }
-    },
-  }));
+  useImperativeHandle(ref, () => {
+    return {
+      list: containerRef.current,
+      focus: options => {
+        const shouldFocusKey =
+          mergedActiveKey ??
+          childList.find(
+            node =>
+              !node.props.disabled &&
+              (node.props.children.length || node.props.children.type),
```

Comment:
```suggestion
              (node.props.children.length || node.props.children.type === 'group'),
```

Reply:
@afc163 Thank you for suggestion, but after some testing we found out with @yhy-1 that current solution doesn't work properly as the focus moves to the group (which is not focusable), instead of the first focusable child of this group. Currenlty I am working on improvements to fix it. Will push my changes soon.